### DISCUSSION TITLE
top.tangyh.lamp.OauthServerApplication  服务启动报错

### DISCUSSION THREAD

**Discussion by 923508946** [zuihou/zuihou-admin-cloud#330] [01.常见问题]
Spring Boot Version: 3.1.3
当前服务: lamp-oauth-server   激活环境: dev   当前版本: 3.11.0-java17

Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
[lamp-oauth-server:8773::] 2024-03-01 09:11:42.354:[ERROR] [37028] [] [main:10528] [o.s.b.diagnostics.LoggingFailureAnalysisReporter.report:40] 

***************************
APPLICATION FAILED TO START
***************************

Description:

Parameter 0 of constructor in top.tangyh.basic.validator.controller.FormValidatorController required a bean of type 'top.tangyh.basic.validator.extract.IConstraintExtract' that could not be found.

Action:

Consider defining a bean of type 'top.tangyh.basic.validator.extract.IConstraintExtract' in your configuration.

[lamp-oauth-server:8773::] 2024-03-01 09:11:42.355:[ WARN] [37028] [] [Thread-1:10529] [c.alibaba.nacos.common.http.HttpClientBeanHolder.shutdown:102] [HttpClientBeanHolder] Start destroying common HttpClient
[lamp-oauth-server:8773::] 2024-03-01 09:11:42.356:[ WARN] [37028] [] [Thread-7:10530] [com.alibaba.nacos.common.notify.NotifyCenter.shutdown:136] [NotifyCenter] Start destroying Publisher
[lamp-oauth-server:8773::] 2024-03-01 09:11:42.356:[ WARN] [37028] [] [Thread-7:10530] [com.alibaba.nacos.common.notify.NotifyCenter.shutdown:153] [NotifyCenter] Destruction of the end

进程已结束，退出代码为 1

![1709256328983](https://github.com/dromara/lamp-cloud/assets/75707148/8f7053a2-0c24-4215-9ad3-072271c97d19)

**Comment by zuihou** [comment]
检查nacos配置是否正确，程序是否能正常读取nacos中的配置

**Reply by 923508946** [reply]
qing'wen请问如何确认是否正常读取呢？第一次使用nacos
![1709257325538](https://github.com/dromara/lamp-cloud/assets/75707148/961d4f1a-d034-4a8a-9d9b-728b6646aa98)

**Reply by 923508946** [reply]
能够读取nacos配置，当我将lamp.rabbitmq.enabled 设置为true 时控制台的提示语就不打印了......

![1709259515525](https://github.com/dromara/lamp-cloud/assets/75707148/290ad49d-d17a-4307-bc1d-a8801839d0c3)

![1709259583370](https://github.com/dromara/lamp-cloud/assets/75707148/f4cd557e-a91b-4f34-b83d-d68f7c93f3d1)

**Comment by zuihou** [comment]
nacos 里面有多少配置文件，介个图

**Reply by 923508946** [reply]
![1709260090571](https://github.com/dromara/lamp-cloud/assets/75707148/9f6e7812-e31a-4114-9125-297bdffb54b6)

**Reply by 923508946** [reply]
目前只有这一个服务无法启动，报错不变
![1709264574430](https://github.com/dromara/lamp-cloud/assets/75707148/42898424-1ad3-422f-827c-a372618bacb8)

![1709264595714](https://github.com/dromara/lamp-cloud/assets/75707148/29a35b34-c1cd-405e-b89d-04d2c665d998)

**Reply by zuihou** [reply]
嗯嗯，我本地排查下

**Reply by zuihou** [reply]
启动类少了注解： `@EnableFormValidator`


<img width="737" alt="image" src="https://github.com/dromara/lamp-cloud/assets/8500410/8441622c-9850-4800-a117-d5a15f1a112a">

**Reply by zuihou** [reply]
加上去试试

**Reply by 923508946** [reply]
启动成功！非常感谢！
